spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -3053,4 +3050,15 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
     }
 
   }
+
+  // TODO: Remove this constraint when we upgrade to new arrow-rs including
+  // https://github.com/apache/arrow-rs/pull/6225
+  def supportedSortType(op: SparkPlan, sortOrder: Seq[SortOrder]): Boolean = {
+    if (sortOrder.length == 1 && sortOrder.head.dataType.isInstanceOf[StructType]) {
+      withInfo(op, "Sort on single struct column is not supported")
```

Comment:
As @andygrove mentioned in #811, there are some more types that are not supported. I will add them in other PR.